### PR TITLE
[WIP]: Add alertmanager.yaml

### PR DESCRIPTION
**Pull request by zgalor**
Add a template that creates a pod with alertmanager behind oauth-proxy

### CODE REVIEW

File: examples/prometheus/alertmanager.yaml [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  metadata:
+    name: alertmanager
+    namespace: "${NAMESPACE}"
+  spec:
+    host: "${ALERTMANAGER_HOSTNAME}"
+    to:
+      name: alertmanager
+    tls:
+      termination: Reencrypt
+      destinationCACertificate: "${SERVICE_SIGNER_CERT}"
```
Comment (smarterclayton) [inline examples/prometheus/alertmanager.yaml]: Please do not use this construct here either.

File: examples/prometheus/alertmanager.yaml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+parameters:
+- description: The namespace to instantiate alertmanager under. Defaults to 'kube-system'.
+  name: NAMESPACE
+  value: kube-system
+- description: The location of the proxy image
+  name: IMAGE_PROXY
+  value: openshift/oauth-proxy:v1.0.0
+- description: The location of the alertmanager image
+  name: IMAGE_ALERTMANAGER
+  value: prom/alertmanager
```
Comment (smarterclayton) [inline examples/prometheus/alertmanager.yaml]: If upstream uses versioned tags, do that here.

File: examples/prometheus/alertmanager.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: alertmanager
+  annotations:
+    "openshift.io/display-name": alertmanager
+    description: |
+      A monitoring solution for an OpenShift cluster - collect and gather metrics from nodes, services, and the infrastructure.
```
Comment (smarterclayton) [inline examples/prometheus/alertmanager.yaml]: Update the description to be accurate.

File: examples/prometheus/alertmanager.yaml (hunk 2 above)
Comment (zgalor) [inline examples/prometheus/alertmanager.yaml]: Is an openshift fork planned for alertmanager? prometheus-alert-buffer?

Comment (smarterclayton) [inline examples/prometheus/alertmanager.yaml]: For buffer yes, for alert manager eventually.

**Comment by zgalor** [comment]
Opened  https://github.com/openshift/origin/pull/15078, that contains an extended template with prometheus pod and message buffer.
closing this to avoid confusion.

**Pull request closed without merging**